guide/samples/src/lro.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: use
+use gcp_sdk_longrunning as longrunning;
+use google_cloud_speech_v2 as speech;
+// ANCHOR_END: use
+
+// ANCHOR: start
+pub async fn start(project_id: &str) -> crate::Result<()> {
+    // ANCHOR: client
+    let client = speech::client::Speech::new().await?;
+    // ANCHOR_END: client
+
+    // ANCHOR: request-builder
+    let operation = client
+        .batch_recognize(format!(
+            "projects/{project_id}/locations/global/recognizers/_"
+        ))
+        // ANCHOR_END: request-builder
+        // ANCHOR: audio-file
+        .set_files([speech::model::BatchRecognizeFileMetadata::new()
+            .set_uri("gs://cloud-samples-data/speech/hello.wav")])
+        // ANCHOR_END: audio-file
+        // ANCHOR: transcript-output
+        .set_recognition_output_config(
+            speech::model::RecognitionOutputConfig::new()
+                .set_inline_response_config(speech::model::InlineOutputConfig::new()),
+        )
+        // ANCHOR_END: transcript-output
+        // ANCHOR: configuration
+        .set_processing_strategy(
+            speech::model::batch_recognize_request::processing_strategy::DYNAMIC_BATCHING,
+        )
+        .set_config(
+            speech::model::RecognitionConfig::new()
+                .set_language_codes(["en-US"])
+                .set_model("short")
+                .set_auto_decoding_config(speech::model::AutoDetectDecodingConfig::new()),
+        )
+        // ANCHOR_END: configuration
+        // ANCHOR: send
+        .send()
+        .await?;
+    // ANCHOR_END: send
+    println!("LRO started, response={operation:?}");
+
+    // ANCHOR: call-manual
+    let response = manually_poll_lro(client, operation).await;
+    println!("LRO completed, response={response:?}");
+    // ANCHOR_END: call-manual
+
+    Ok(())
+}
+// ANCHOR_END: start
+
+// ANCHOR: automatic
+pub async fn automatic(project_id: &str) -> crate::Result<()> {
+    // ANCHOR: automatic-use
+    use speech::Poller;
```

Comment:
off topic: I am not sure it is a good idea or not, but I know some crates have a prelude of there own. If we end up needing a bunch of other imports to make functionality work, we could consider having something similar.